components/enterprise-contract/download-service.yaml

```diff
@@ -0,0 +1,234 @@
+      initContainers:
+        - args:
+            - |
+              #!/usr/bin/bash
+
+              set -o errexit
+              set -o nounset
+              set -o pipefail
+
+              function handle_error {
+                sleep infinity
+              }
+              trap handle_error ERR
+
+              imgref=$({
+                cd "$(mktemp -d)"
+                oc image extract "$EC_TASK_BUNDLE_REF"
+                jq -r '.spec.steps[0].image' verify-enterprise-contract
+              })
+
+              base="${imgref%:*}"
+              base="${imgref%@*}"
+
+              manifests="$(oc image info --output=json --show-multiarch=true "${imgref}")"
+
+              download_ec() {
+                os="$1"
+                architecture="$2"
+                digest="$3"
+
+                mkdir -p "${os}_${architecture}"
+                oc image extract "${base}@${digest}" --path /usr/bin/ec:"${os}_${architecture}" --confirm
+                chmod +x "${os}_${architecture}/ec"
+                tar czf "${os}_${architecture}/ec_${os}_${architecture}.tar.gz" -C "${os}_${architecture}" ec
+              }
+
+              while read -r line; do
+                read -r -a args <<< "${line}"
+                download_ec "${args[@]}" &
+              done < <(echo "${manifests}" | jq -r ".[] | \"\(.config.os) \(.config.architecture) \(.digest)\"")
+
+              # shellcheck disable=SC2046
+              wait $(jobs -p)
+
+              ingress="$(oc get route ec-cli-download-route --output=jsonpath='{.spec.host}')"
+
+              {
+                echo "apiVersion: console.openshift.io/v1
+              kind: ConsoleCLIDownload
+              metadata:
+                name: ec-cli
+              spec:
+                description: |
+                  The \`ec\` CLI is used to evaluate Enterprise Contract policies for Software Supply Chain.
+                displayName: ec - Enterprise Contract CLI
+                links:"
+                echo "${manifests}" | jq -r ".[] | \"    - href: https://${ingress}/\(.config.os)_\(.config.architecture)/ec_\(.config.os)_\(.config.architecture).tar.gz
+                    text: Download ec CLI for \(.config.os)/\(.config.architecture)\""
+              } | oc apply -f -
+
+              {
+                echo '<!doctype html>
+                <html>
+                <style>
+                html { font-family: "RedHatText", "Overpass", overpass, helvetica, arial, sans-serif; line-height: 1.5rem; }
+                ul { list-style: none; padding-left: 0; }
+                </style>
+                <body>
+                <ul>'
+                echo "${manifests}" | jq -r ".[] | \"<li><a href=\\\"\(.config.os)_\(.config.architecture)/ec_\(.config.os)_\(.config.architecture).tar.gz\\\">Download ec CLI for \(.config.os)/\(.config.architecture)</a></li>\""
+                echo '</ul>
+                </body>
+                </html>'
+              } > index.html
```

Comment:
It feels a bit untidy to have all the echo ... | oc apply and and html created in here. Would it be nicer to make a little custom image for the httpd server? Just an idea, not a blocker.

Ps, does this resemble the way that TAS does it for their cli downloads?

Reply:
My main goal here was not to produce a new image at all. From what I understand RHTAS builds [an image](https://github.com/securesign/sigstore-ocp/blob/914bdbc5f29c40dcc8fc73989dd3561663324758/images/Dockerfile-clientserver) with the CLIs baked in.

Reply:
It looks like they provide CLI binaries for Windows, Mac, and Linux for x86_64(amd64). I don't think this approach is going to work for the Windows and Mac CLIs. :disappointed: 

Linux ppc64le and s390x are really low in their priority list. They may never provide those.

I think we should follow suit.

Reply:
FWIW I think it's okay if that comes in a future PR.